app/code/Magento/InventorySourceSelectionApi/Model/GetAvailableSourceItemsBySkusAndSortedSource.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventorySourceSelectionApi\Model;
+
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\InventoryApi\Api\Data\SourceItemInterface;
+use Magento\InventoryApi\Api\SourceItemRepositoryInterface;
+
+/**
+ * Retrieve source items for a defined set of skus and sorted source codes
+ *
+ * Useful for determining availability and source selection
+ */
+class GetAvailableSourceItemsBySkusAndSortedSource
+{
+    /**
+     * @var SourceItemRepositoryInterface
+     */
+    private $sourceItemRepository;
+
+    /**
+     * @var SearchCriteriaBuilder
+     */
+    private $searchCriteriaBuilder;
+
+    /**
+     * @param SourceItemRepositoryInterface $sourceItemRepository
+     * @param SearchCriteriaBuilder $searchCriteriaBuilder
+     * @SuppressWarnings(PHPMD.LongVariable)
+     */
+    public function __construct(
+        SourceItemRepositoryInterface $sourceItemRepository,
+        SearchCriteriaBuilder $searchCriteriaBuilder
+    ) {
+        $this->sourceItemRepository = $sourceItemRepository;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute(array $skus, array $sortedSourceCodes): array
+    {
+        $searchCriteria = $this->searchCriteriaBuilder
+            ->addFilter(SourceItemInterface::SKU, $skus, 'in')
+            ->addFilter(SourceItemInterface::SOURCE_CODE, $sortedSourceCodes, 'in')
+            ->addFilter(SourceItemInterface::QUANTITY, 0, 'gt')
```

Comment:
Is it really needed to add QTY filter here? In stock below seems to be sufficient

Reply:
Is this also valid with backorders?

Reply:
No, it is not needed. We will have one extra loop in `\Magento\InventorySourceSelectionApi\Model\GetInStockSourceItemsBySkusAndSortedSource::execute` for each product with zero quantity and in stock. Adding this filter at MySQL level could be slightly faster, but it is probably out of its responsability to check the quantity, so I agree removing it.